IBAnimatable/MaskDesignable.swift

```diff
@@ -6,47 +6,34 @@
 import UIKit
 
 public protocol MaskDesignable {
-  var maskType: String? { get set }
+  var maskType: MaskType { get set }
 }
 
 public extension MaskDesignable where Self: UIView {
+  
   public func configMask() {
-    guard let unwrappedMaskType = maskType else {
-      return
-    }
     
-    if let rawMaskType = MaskType(rawValue: unwrappedMaskType) {
-      switch rawMaskType {
-      case .circle:
-        maskCircle()
-      case .star:
-        maskStar()
-      case .polygon:
-        maskPolygon()
-      case .triangle:
-        maskTriangle()
-      case .wave:
-        maskWave()
-      case .parallelogram:
-        maskParallelogram()
-      }
-    } else {
-      if unwrappedMaskType.hasPrefix(MaskType.star.rawValue) {
-        maskStarFromString(unwrappedMaskType)
-      } else if unwrappedMaskType.hasPrefix(MaskType.wave.rawValue) {
-        maskWaveFromString(unwrappedMaskType)
-      } else if unwrappedMaskType.hasPrefix(MaskType.polygon.rawValue) {
-        maskPolygonFromString(unwrappedMaskType)
-      } else if unwrappedMaskType.hasPrefix(MaskType.parallelogram.rawValue) {
-        maskParallelogramFromString(unwrappedMaskType)
-      }
-      
+    switch maskType {
+    case .circle:
+      maskCircle()
+    case .parallelogram(let angle):
+      maskParallelogram(angle ?? 60)
+    case .polygon(let sides):
+      maskPolygon(sides ?? 6)
+    case .star(let points):
+      maskStar(points ?? 5)
+    case .wave(let direction, let width, let offset):
+      maskWave(direction ?? .up, waveWidth: width ?? 40.0, waveOffset:offset ?? 0.0)
```

Comment:
Already comment this somewhere else, but I definitely think it will be worth to set the default value in the enums instead of here. It will be easier to get what is the default!